pypeit/edgetrace.py

```diff
@@ -0,0 +1,4122 @@
+            msgs.info('No traces defined.')
+            plt.xlim(-img_buffer, self.nspec+img_buffer)
+            plt.ylim(-img_buffer, self.nspat+img_buffer)
+            plt.xlabel('Spectral pixel index')
+            plt.ylabel('Spatial pixel index')
+            plt.show()
+            return
+
+        # Build the data to plot. Any masked data is excluded; masked
+        # data to be plotted are held in a separate array. This means
+        # that errors and fits are currently never plotted for masked
+        # data.
+        _flag = None if flag in ['any', None] else np.atleast_1d(flag)
+        cen = np.ma.MaskedArray(self.spat_cen, mask=self.bitmask.flagged(self.spat_msk))
+        err = np.ma.MaskedArray(self.spat_err, mask=np.ma.getmaskarray(cen).copy())
+        msk = None if flag is None \
+                    else np.ma.MaskedArray(self.spat_cen, mask=np.invert(
+                                            self.bitmask.flagged(self.spat_msk, flag=_flag)))
+
+        # Select the traces to show
+        trace_indx = np.ones(self.ntrace, dtype=bool) if traceid is None \
+                        else np.isin(self.traceid, np.atleast_1d(traceid))
+
+        # Spectral position
+        spec = np.tile(np.arange(self.nspec), (self.ntrace,1)).T
+
+        if include_error:
+            # Show the errors
+            plt.errorbar(spec[::thin,trace_indx].ravel(), cen[::thin,trace_indx].ravel(),
+                         yerr=err[::thin,trace_indx].ravel(), fmt='none', ecolor='k',
+                         elinewidth=0.5, alpha=0.3, capthick=0, zorder=3)
+
+        # Plot the left trace points
+        left = (self.traceid < 0) & trace_indx
+        plt.scatter(spec[::thin,left], cen[::thin,left], marker='.', color='k', s=30,
+                    lw=0, zorder=4, label='left edge measurements', alpha=0.8)
+        if msk is not None:
+            plt.scatter(spec[::thin,left], msk[::thin,left], marker='x', color='C3', s=20,
+                        lw=0.5, zorder=5, label='masked left edges', alpha=0.8)
+
+        # Plot the right trace points
+        right = (self.traceid > 0) & trace_indx
+        plt.scatter(spec[::thin,right], cen[::thin,right], marker='.', color='0.7',
+                    s=30, lw=0, zorder=4, label='right edge measurements', alpha=0.8)
+        if msk is not None:
+            plt.scatter(spec[::thin,right], msk[::thin,right], marker='x', color='C1', s=20,
+                        lw=0.5, zorder=5, label='masked right edges', alpha=0.8)
+
+        if idlabel:
+            # Label the traces by their ID number
+            for i in range(self.ntrace):
+                if not trace_indx[i]:
+                    continue
+                indx = np.invert(cen.mask[:,i])
+                if not np.any(indx):
+                    continue
+                _spec = spec[:,i][indx]
+                _cen = cen[:,i][indx]
+                plt.text(_spec[_spec.size//2], _cen[_spec.size//2], str(self.traceid[i]),
```

Comment:
Can you explain why you tend to unpack these as the start of these methods?
I rather like seeing them with `par` attached so that I know they can be set.

Reply:
Yeah, this was just an artifact of the development process.  I starting by having everything be arguments, and then decided what arguments should be in the parameter list.  Instead of rewriting the code then to use the `par` stuff explicitly, I just pulled what I needed from the parameters at the beginning.  Generally, I'd rather we be using the `par` directly instead of passing off to a middle man.  I'm actually a bit torn though between that and liking the clarity  of collecting what parameters are used at the beginning of each method...

Reply:
I understand the complaint about the variable and parameter names being different.  It was more clear to me to have the name of the variable match the keyword argument of the relevant function instead of the parameter name.  Another useful aspect of collecting all the variables extracted from `par` is you can see the variable to parameter mapping.  I'd rather make the effort to use `par` directly, but I'm leaving this for now.